contracts/src/factory/facets/partner/PartnerRegistryBase.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+// interfaces
+import {IPartnerRegistryBase} from "./IPartnerRegistry.sol";
+
+// libraries
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {PartnerRegistryStorage} from "./PartnerRegistryStorage.sol";
+
+// contracts
+
+abstract contract PartnerRegistryBase is IPartnerRegistryBase {
+  using EnumerableSet for EnumerableSet.AddressSet;
+
+  bytes32 internal constant CURRENT_VERSION = keccak256("1");
+
+  function __PartnerRegistryBase_init(
+    uint256 registryFee,
+    uint256 maxPartnerFee
+  ) internal {
+    PartnerRegistryStorage.Layout storage ds = PartnerRegistryStorage.layout();
+
+    ds.partnerSettingsByVersion[CURRENT_VERSION] = PartnerRegistryStorage
```

Comment:
I'm not sure what we're trying to achieve with CURRENT_VERSION. This data storage is mutable though I intuitively think that what is trying to be achieved (correct me) is an immutable data store of versioned partner registry settings. If we update the implementation of PartnerRegistry (facet upgrade) but want to use some prior version, we should be able to pass in that version and check that it exists, and then use it if so right ? 

Also, is the intent to be able to support different versions per partner ? If not, I'm not sure I understand why we need to store multiple versions of settings in this map at all.

Reply:
right, this can be removed